rewrite/rewrite/python/format/spaces_visitor.py

```diff
@@ -0,0 +1,166 @@
+        m = m.padding.with_arguments(m.padding.arguments.with_before(
+            Space.SINGLE_SPACE if self._style.before_parentheses.method_call else Space.EMPTY))
+
+        if not m.arguments or isinstance(m.arguments[0], Empty):
+            # Handle within method call parenthesis with no arguments e.g. foo() <-> foo( )
+            use_space = self._style.within.empty_method_call_parentheses
+            m = m.padding.with_arguments(
+                m.padding.arguments.padding.with_elements(
+                    [arg.with_element(self.space_before(arg.element, use_space)) for arg in
+                     m.padding.arguments.padding.elements]
+                )
+            )
+        else:
+            # Handle:
+            # - Within method call parenthesis with arguments e.g. foo( 1, 2 ) <-> foo(1, 2).
+            #   Note: this only applies to the space left of the first argument and right of the last argument.
+            # - Space after comma e.g. foo(1,2,3) <-> foo(1, 2, 3)
+            args_size = len(m.arguments)
+            use_space = self._style.within.method_call_parentheses
+
+            def _process_argument(index, arg, args_size, use_space):
+                if index == 0:
+                    arg = arg.with_element(self.space_before(arg.element, use_space))
+                else:
+                    arg = arg.with_element(self.space_before(arg.element, self._style.other.after_comma))
+
+                if index == args_size - 1:
+                    arg = self.space_after(arg, use_space)
+                else:
+                    arg = self.space_after(arg, self._style.other.before_comma)
+
+                return arg
+
+            m = m.padding.with_arguments(
+                m.padding.arguments.padding.with_elements(
+                    [_process_argument(index, arg, args_size, use_space)
+                     for index, arg in enumerate(m.padding.arguments.padding.elements)]
+                )
+            )
+        # TODO: Handle type parameters, relevant for constructors in Python.
+        return m
+
+    def visit_array_access(self, array_access: ArrayAccess, p: P) -> J:
+        a: ArrayAccess = cast(ArrayAccess, super().visit_array_access(array_access, p))
+        use_space_within_brackets = self._style.within.brackets
+        index_padding = a.dimension.padding.index
+        element_prefix = update_space(index_padding.element.prefix, use_space_within_brackets)
+        index_after = update_space(index_padding.after, use_space_within_brackets)
+
+        a = a.with_dimension(
+            a.dimension.padding.with_index(
+                index_padding.with_element(
+                    index_padding.element.with_prefix(element_prefix)
+                ).with_after(index_after)
+            ).with_prefix(update_space(a.dimension.prefix, self._style.before_parentheses.left_bracket))
+        )
+        return a
+
+    def visit_assignment(self, assignment: Assignment, p: P) -> J:
```

Comment:
For chained assignments I had to add a new `Py.ChainedAssignment` LST type, so you will here have to also override `visit_chained_assignment()`.

Reply:
Should be fixed now.